tests/tests_full_integration_dc.yml

```
@@ -0,0 +1,107 @@
+#  ad_integration_join_to_dc="ad1.domain.com" \
+#  ad_integration_password=Secret123 ad_integration_realm="domain.com"
+# [ad]
+# ad1 ansible_host=<AD IP> ansible_connection=winrm ansible_password=Secret123\
+#  ansible_port=5986 ansible_user=Administrator \
+#  ansible_winrm_server_cert_validation=ignore
+
+- name: Set variables
+  hosts: client
+  tasks:
+    - name: Set ad_integration_realm if it is not defined or is empty
+      set_fact:
+        ad_integration_realm: domain.com
+      when: ad_integration_realm is not defined or not ad_integration_realm
+
+    - name: Set ad_integration_join_to_dc if it is not defined or is empty
+      set_fact:
+        ad_integration_join_to_dc: "{{ groups.ad | first }}"
+      when:
+        - ad_integration_join_to_dc is not defined
+          or not ad_integration_join_to_dc
+
+- name: Prepare user and groups on the AD
+  hosts: ad
+  tasks:
+    - name: Create groups
+      win_domain_group:
+        name: "{{ item.name }}"
+        state: present
+        scope: "{{ item.scope }}"
+      loop:
+        - { name: 'test_grp0', scope: 'universal' }
+        - { name: 'test_grp1', scope: 'universal' }
+      register: group_creation
+    - debug:
+        var: group_creation
+
+    - name: Add a test user
+      win_domain_user:
+        name: "{{ item.name }}"
+        firstname: "{{ item.first_name }}"
+        surname: "{{ item.surname }}"
+        password: Secret123
+        state: present
+        groups:
+          - test_grp1
+      loop:
+        - { name: 'test_usr1', first_name: 'Josef', surname: 'Novak' }
+      register: user_creation
+    - debug:
+        var: user_creation
+
+- name: Ensure that the role runs with real AD - join to DC
+  hosts: client
+  tasks:
+    - name: Run tests
+      block:
+
+        - name: Test run the system role
+          include_role:
+            name: linux-system-roles.ad_integration
+          register: role_run
+        - debug:
+            var: role_run
+
+        - name: Restart sssd
+          service:
+            name: sssd
+            state: restarted
+
+        - name: Test sssd config contains the realm
+          command:
+            cmd: 'cat /etc/sssd/sssd.conf'
+          register: sssd_config
+          failed_when: '"{{ ad_integration_realm }}" not in sssd_config.stdout'
+
+        - name: Test joined realm
+          shell:
+            cmd: realm list
+          register: realm_list
+          failed_when: '"{{ ad_integration_realm }}" not in realm_list.stdout'
+        - debug:
+            var: realm_list
+
+        - name: Test verify user is present
+          shell:
+            cmd: getent passwd test_usr1@{{ ad_integration_realm }}
+
+        - name: Test verify group is present
+          shell:
+            cmd: getent group test_grp1@{{ ad_integration_realm }}
+
+      always:
+        - name: Teardown - Leave realm
+          shell: cmd="realm leave"
+          ignore_errors: yes
```

```suggestion
          ignore_errors: true  # noqa ignore-errors
```
This will make the new version of ansible-lint happy

---

Please use `true`/`false` instead of `yes`/`no`

---

Done